build

```diff
@@ -1,10 +1,29 @@
 #!/bin/bash
+#
+# Build a new Python Docker image.
+#
+# Usage:
+#   ./build [--gpu] [--use-cache]
+#
+# Options:
+#   --gpu: Build an image with GPU support.
+#   --use-cache: Use layer cache when building a new image.
+# 
 set -e
+set -x
+
+CACHE_FLAG="--no-cache"
+IMAGE_TAG="kaggle/python-build"
+DOCKERFILE="Dockerfile"
+
+if [[ "$1" == "--gpu" ]]; then
+    IMAGE_TAG="kaggle/python-gpu-build"
+    DOCKERFILE="gpu.Dockerfile"
+    shift
+fi
 
-# Default behavior is to do everything from scratch.
-# The --use-cache option is useful if you're iterating on a broken build.
 if [[ "$1" ==  "--use-cache" ]]; then
-    docker build --rm -t kaggle/python-build .
-else
-    docker build --pull --rm --no-cache -t kaggle/python-build .
+    CACHE_FLAG=""
 fi
+
+docker build --rm $CACHE_FLAG -t $IMAGE_TAG -f $DOCKERFILE .
```

Comment:
While you may not be expecting spaces in these values, it is safer to just double quote these (`--rm "$CACHE_FLAG"`).  I typically don't bother with braces (`--rm "${CACHE_FLAG}"`) unless the string contains anything but a single variable, but you may prefer doing that for similar reasons as the quoting.

Reply:
`$CACHE_FLAG` maybe be empty so I didn't put quotes on purpose. I added quotes to the image name/tag even though having a space would be invalid.